rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -256,34 +318,23 @@ def reduce_asset_amount(self, asset: Asset, amount: FVal, timestamp: Timestamp)
         This function does the same as calculate_spend_cost_basis as far as consuming
         acquisitions is concerned but does not calculate bought cost.
         """
-        # No need to do anything if amount is to be reduced by zero
-        if amount == ZERO:
-            return True
-
         asset_events = self.get_events(asset)
-        if len(asset_events.acquisitions) == 0:
+        if len(asset_events.acquisitions_manager.get_acquisitions()) == 0:
```

Comment:
Better than consuming + creating a tuple for the deque just to check the length why you don't create the `__len__` method in the `CostBasisEvents` class where you do the `len(deque)`?

Reply:
1) I guess you meant `BaseAcquisitionsOrder` since `CostBasisEvents` has 3 properties, not only `acquisitions`, right?
2) Won't it decrease readability? since `acquisitions_manager` is actually a `manager` for acquisitions and is not something iterable